www/src/pages/api/login.ts

```diff
@@ -0,0 +1,57 @@
+import type { APIRoute } from 'astro';
+import { z } from 'zod';
+import { User } from '@backend/user';
+
+const loginErrors = {
+    email: {
+        empty: 'Email cannot be empty',
+        invalid: 'Enter valid email',
+    },
+    password: 'Password cannot be empty',
+    form: 'Invalid credentials',
+} as const;
+
+const schema = z.object({
+    email: z.string().nonempty({ message: loginErrors.email.empty }).email({ message: loginErrors.email.invalid }),
+    password: z.string().nonempty({ message: loginErrors.password }),
+});
+
+export const POST: APIRoute = async ({ locals }) => {
+    const { jsonData, req } = locals;
+    const parseResult = schema.safeParse(jsonData);
+
+    if (!parseResult.success) {
+        const errors = parseResult.error.flatten().fieldErrors;
+
+        return new Response(JSON.stringify({ errors }), {
+            status: 400,
+            headers: { 'Content-Type': 'application/json' },
+        });
```

Comment:
Please use `Reponse.json` instead of manually creating a Response object.

```suggestion
        return Response.json({ errors }, { status: 400 });
```

Reply:
Good point.